pkg/controller/manager/manager_controller.go

```diff
@@ -650,6 +655,62 @@ func (r *ReconcileManager) processVRouters(manager *v1alpha1.Manager, replicas *
 	return nil
 }
 
+func (r *ReconcileManager) processContrailCNIs(manager *v1alpha1.Manager) error {
+	for _, existingContrailCNI := range manager.Status.ContrailCNIs {
+		found := false
+		for _, intendedContrailCNI := range manager.Spec.Services.ContrailCNIs {
+			if *existingContrailCNI.Name == intendedContrailCNI.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			oldContrailCNI := &v1alpha1.ContrailCNI{}
```

Comment:
Shouldn't we filter these objects by labels, so that only contrailcni's belonging to the current manager will be removed?

```
labels:
    contrail_cluster: cluster1
```

Reply:
Done